src/service.rs

```diff
@@ -116,28 +130,71 @@ impl Service {
-            &mut buf,
-        )?;
+        fn encode(arg: impl prost::Message) -> Vec<u8> {
+            let mut buf = Vec::new();
+            prost::Message::encode(&arg, &mut buf).unwrap();
+            buf
+        }
 
         let config = prost_types::Any {
             type_url: "type.googleapis.com/envoy.extensions.filters.http.router.v3.Router"
                 .to_string(),
-            value: buf,
+            value: encode(Router {
+                ..Default::default()
+            }),
+        };
+
+        // WASM section, @TODO move out to a new method
+        let wasm_filter = Wasm {
+            config: Some(PluginConfig {
+                name: format!("Service::{:?}", self.id),
+                root_id: format!("Service::{:?}", self.id),
+                vm: Some(Vm::VmConfig(VmConfig {
+                    vm_id: format!("Service::{:?}", self.id),
+                    runtime: "envoy.wasm.runtime.v8".to_string(),
+                    configuration: Some(prost_types::Any {
+                        type_url: "type.googleapis.com/google.protobuf.StringValue".to_string(),
+                        value: encode(serde_json::to_string(&self.clone()).unwrap()),
+                    }),
+                    code: Some(AsyncDataSource {
+                        specifier: Some(Specifier::Remote(RemoteDataSource {
+                            http_uri: Some(HttpUri {
+                                uri: format!("http://control-plane:5001/{}", WASM_FILTER_PATH),
+                                timeout: Some(Duration {
+                                    seconds: 100,
+                                    nanos: 0,
+                                }),
+                                http_upstream_type: Some(HttpUpstreamType::Cluster(
+                                    "wasm_files".to_string(),
+                                )),
+                            }),
+                            sha256: self.get_wasm_filter_sha().unwrap(),
+                            ..Default::default()
+                        })),
+                    }),
+                    ..Default::default()
+                })),
+                ..Default::default()
+            }),
         };
 
         let connection_manager = HttpConnectionManager {
             stat_prefix: "ingress_http".to_string(),
             codec_type: 0,
-            http_filters: vec![HttpFilter {
-                name: "envoy.filters.http.router".to_string(),
-                config_type: Some(http_filter::ConfigType::TypedConfig(config)),
-            }],
+            http_filters: vec![
+                HttpFilter {
+                    name: "envoy.filters.http.wasm".to_string(),
+                    config_type: Some(http_filter::ConfigType::TypedConfig(prost_types::Any {
+                        type_url: "type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm"
+                            .to_string(),
+                        value: encode(wasm_filter),
```

Comment:
Since we are calling `encode` on the whole `wasm_filter`, do we need to call `encode` above on the stringified JSON?

Reply:
post::Any need to have vec<u8> so encode is the best way here.